server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexStateServiceBatchingTests.java

```diff
@@ -98,8 +100,11 @@ public void testBatchCloseIndices() throws Exception {
         createIndex("test-3", indicesAdmin().prepareCreate("test-3"));
         ensureGreen("test-1", "test-2", "test-3");
 
-        final var assertingListener = closedIndexCountListener(3);
-        clusterService.addListener(assertingListener);
+        final List<String[]> observedClosedIndices = new ArrayList<>();
```

Comment:
Does it make sense to declare the list as `Collections.synchronizedList`, because the observer can be called concurrently?

Reply:
Right. I updated it to Collections.synchronizedList. Thanks!